apps/predictions/lib/predictions_pub_sub.ex

```diff
@@ -71,6 +78,37 @@ defmodule Predictions.PredictionsPubSub do
     {:noreply, state}
   end
 
+  def handle_info(
+        {:DOWN, parent_ref, :process, caller_pid, _reason},
+        state
+      ) do
+    Process.demonitor(parent_ref, [:flush])
+    {key, new_state} = Map.pop(state, caller_pid)
+    # Here we can check if there are other subscribers for the associated key.
+    # If there are no other subscribers remaining, we stop the associated
+    # predictions data stream.
+    if other_subscribers(key, caller_pid) == [] do
+      StreamSupervisor.stop_stream(key)
+    end
+
+    {:noreply, new_state}
+  end
+
+  # find registrations for this key from processes other than the indicated pid
+  defp other_subscribers(key, pid_to_omit) do
+    registry_key = self()
+    pattern = {:"$1", :"$2", :"$3"}
+
+    guards = [
+      {:==, :"$1", registry_key},
+      {:"=/=", :"$2", pid_to_omit},
+      {:==, :"$3", key}
+    ]
+
+    body = [{{:"$2", :"$3"}}]
+    Registry.select(@subscribers, [{pattern, guards, body}])
+  end
```

Comment:
suggestion(non-blocking):
```suggestion
  defp no_other_subscribers?(key, pid_to_omit) do
    registry_key = self()
    pattern = {registry_key, :"$2", key}

    guards = [
      {:"=/=", :"$2", pid_to_omit},
    ]

    body = [true]
    Registry.select(@subscribers, [{pattern, guards, body}]) == []
  end
```

Since you don't care about the specific values from the Registry, you can:
- move registry and key matches into the pattern
- return a random value as the body, rather than any particular values